compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/mapping/IrTypeMapper.kt

```diff
@@ -128,7 +128,8 @@ open class IrTypeMapper(private val context: JvmBackendContext) : KotlinTypeMapp
     open fun mapType(
         type: IrType,
         mode: TypeMappingMode = TypeMappingMode.DEFAULT,
-        sw: JvmSignatureWriter? = null
+        sw: JvmSignatureWriter? = null,
+        materialized: Boolean = true
     ): Type = AbstractTypeMapper.mapType(this, type, mode, sw)
```

Comment:
I did that because `materialized` is currently only used in the override of this method in `ClassCodegen` to avoid inner classes attributes. Passing it into `AbstractTypeMapper.mapType` just means that we will be threading around a boolean in there that is not needed for anything.

I added another commit that does that threading of the unused boolean. That way it is easy to either land just the first commit or both of them depending on what you prefer. :)